restapi/api_client.go

```diff
@@ -211,6 +222,12 @@ func (client *api_client) send_request(method string, path string, data string)
 		log.Printf("%s\n", body)
 	}
 
+	// Rate limiting
+	if client.debug {
+		log.Printf("Waiting for rate limit availability\n")
+	}
+	_ = client.rate_limiter.Wait(context.Background())
```

Comment:
It's not that I distrust this library or anything, but I'd prefer for it to not be in the main logic flow unless a timeout is actually set. Can you adjust to only allocate `rateLimiter` if a timeout is set in the options? Something like this could be used here, then...
```
    // Rate limiting
    if client.rateLimeter != nil {
      if client.debug {
        log.Printf("Waiting for rate limit availability\n")
      }
      _ = client.rate_limiter.Wait(context.Background())
    }
```

Reply:
@DRuggeri I have added the recommended changes in #71